cmd/vela-worker/register.go

```diff
@@ -0,0 +1,36 @@
+// Copyright (c) 2020 Target Brands, Inc. All rights reserved.
+//
+// Use of this source code is governed by the LICENSE file in this repository.
+
+package main
+
+import (
+	"github.com/go-vela/types/library"
+	"github.com/sirupsen/logrus"
+)
+
+// register is a helper function to register
+// the worker in the database, updating the item
+// if the worker already exists
+func (w *Worker) register(config *library.Worker) error {
+	// check to see if the worker already exists in the database
+	_, _, err := w.VelaClient.Worker.Get(config.GetHostname())
+	if err != nil {
```

Comment:
I wonder if it's worth capturing `resp` and doing an additional check to ensure the error that's returned is a `404` to know whether or not to make the call. Right now, your server response could be any error code 4XX or 5XX and you send a call.

Reply:
I think that's a really good idea. I'll add that!

Reply:
Thanks for the suggestion. I also updated the errors in that function to log instead of returning the error. For the time being I don't think we want this to block to workers from being deployed.